<launch>
<!-- Dichiaro i launch arguments -->
<arg name="target_frame" default="flying_sensor" doc="Target frame for the flying sensor" />
<arg name="map_frame" default="map" doc="Map frame for the flying sensor" />
<arg name="init_pose" default="[0,0,2,0,0,0]" doc="Initial pose" />
<arg name="Px" default="5.0" doc="Position Px gain" />
<arg name="Py" default="5.0" doc="Position Py gain" />
<arg name="Pz" default="2.0" doc="Position Pz gain" />
<arg name="Pxdot" default="5.0" doc="Velocity P x gains" />
<arg name="Dxdot" default="0.5" doc="Velocity D x gains" />
<arg name="Ixdot" default="5.0" doc="Velocity I x gains" />
<arg name="Pydot" default="5.0" doc="Velocity P y gains" />
<arg name="Dydot" default="0.5" doc="Velocity D y gains" />
<arg name="Iydot" default="5.0" doc="Velocity I y gains" />
<arg name="Pzdot" default="4.0" doc="Velocity P z gains" />
<arg name="Dzdot" default="0.5" doc="Velocity D z gains" />
<arg name="Izdot" default="5.0" doc="Velocity I z gains" />
<arg name="Pphi" default="4.0" doc="Attitude P phi gain" />
<arg name="Ptheta" default="4.0" doc="Attitude P theta gain" />
<arg name="Ppsi" default="1.5" doc="Attitude P psi gain" />
<arg name="Pp" default="1.5" />
<arg name="Dp" default="0.04" />
<arg name="Pq" default="1.5" />
<arg name="Dq" default="0.04" />
<arg name="Pr" default="1.0" />
<arg name="Dr" default="0.1" />
<arg name="uMax" default="50.0" doc="Max velocity x" />
<arg name="vMax" default="50.0" doc="Max velocity y" />
<arg name="wMax" default="50.0" doc="Max velocity z" />
<arg name="saturateVel_separately" default="true" />
<arg name="tiltMax" default="30.0" />
<arg name="pMax" default="200.0" />
<arg name="qMax" default="200.0" />
<arg name="rMax" default="150.0" />
<arg name="minTotalVel_YawFollow" default="0.1" />
<arg name="useIntegral" default="true" />

    <!-- Launch the node -->
<node pkg="ros2_quad_sim_python" type="ros_quad_sim_and_ctrl.py" name="quadsim" output="screen">
<param name="target_frame" value="$(arg target_frame)" />
<param name="map_frame" value="$(arg map_frame)" />
<param name="init_pose" value="$(arg init_pose)" />
<param name="Px" value="$(arg Px)" />
<param name="Py" value="$(arg Py)" />
<param name="Pz" value="$(arg Pz)" />
<param name="Pxdot" value="$(arg Pxdot)" />
<param name="Dxdot" value="$(arg Dxdot)" />
<param name="Ixdot" value="$(arg Ixdot)" />
<param name="Pydot" value="$(arg Pydot)" />
<param name="Dydot" value="$(arg Dydot)" />
<param name="Iydot" value="$(arg Iydot)" />
<param name="Pzdot" value="$(arg Pzdot)" />
<param name="Dzdot" value="$(arg Dzdot)" />
<param name="Izdot" value="$(arg Izdot)" />
<param name="Pphi" value="$(arg Pphi)" />
<param name="Ptheta" value="$(arg Ptheta)" />
<param name="Ppsi" value="$(arg Ppsi)" />
<param name="Pp" value="$(arg Pp)" />
<param name="Dp" value="$(arg Dp)" />
<param name="Pq" value="$(arg Pq)" />
<param name="Dq" value="$(arg Dq)" />
<param name="Pr" value="$(arg Pr)" />
<param name="Dr" value="$(arg Dr)" />
<param name="uMax" value="$(arg uMax)" />
<param name="vMax" value="$(arg vMax)" />
<param name="wMax" value="$(arg wMax)" />
<param name="saturateVel_separately" value="$(arg saturateVel_separately)" />
<param name="tiltMax" value="$(arg tiltMax)" />
<param name="pMax" value="$(arg pMax)" />
<param name="qMax" value="$(arg qMax)" />
<param name="rMax" value="$(arg rMax)" />
<param name="minTotalVel_YawFollow" value="$(arg minTotalVel_YawFollow)" />
<param name="useIntegral" value="$(arg useIntegral)" />
</node>
</launch>
